views.py

```diff
@@ -42,96 +41,28 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
-        # bag when bag_modified is false due to the flag being out-of-sync with the real bag status
-        if bag_modified is None or bag_modified.lower() == "false":
-            # check whether the bag file exists
-            bag_file_name = res_id + '.zip'
-            if res_root.startswith(res_id):
-                bag_full_path = os.path.join('bags', bag_file_name)
-            else:
-                bag_full_path = os.path.join(federated_path, 'bags', bag_file_name)
-            # set bag_modified to 'true' if the bag does not exist so that it can be recreated
-            # and the bag_modified AVU will be set correctly as well subsequently
-            if not istorage.exists(bag_full_path):
-                bag_modified = 'true'
-        metadata_dirty = istorage.getAVU(res_root, 'metadata_dirty')
-
-    if is_bag_download:
-        # do on-demand bag creation
-        # needs to check whether res_id collection exists before getting/setting AVU on it
-        # to accommodate the case where the very same resource gets deleted by another request
-        # when it is getting downloaded
-
-        # send signal for pre_check_bag_flag
-        pre_check_bag_flag.send(sender=resource_cls, resource=res)
+        raise KeyError('settings must have IRODS_GLOBAL_SESSION set '
+                       'if there is no environment object')
+    if istorage.exists(res_id):
+        bag_modified = istorage.getAVU(res_id, 'bag_modified')
         if bag_modified is None or bag_modified.lower() == "true":
-            if metadata_dirty is None or metadata_dirty.lower() == 'true':
-                create_bag_files(res)
-            if use_async:
-                # task parameter has to be passed in as a tuple or list, hence (res_id,) is needed
-                # Note that since we are using JSON for task parameter serialization, no complex
-                # object can be passed as parameters to a celery task
-                task = create_bag_by_irods.apply_async((res_id,), countdown=3)
-                if rest_call:
-                    return HttpResponse(json.dumps({'bag_status': 'Not ready',
-                                                    'task_id': task.task_id}),
-                                        content_type="application/json")
-
-                request.session['task_id'] = task.task_id
-                request.session['download_path'] = request.path
-                return HttpResponseRedirect(res.get_absolute_url())
-            else:
-                ret_status = create_bag_by_irods(res_id)
-                if not ret_status:
-                    content_msg = "Bag cannot be created successfully. Check log for details."
-                    response = HttpResponse()
-                    if rest_call:
-                        response.content = content_msg
-                    else:
-                        response.content = "<h1>" + content_msg + "</h1>"
-                    return response
+            create_bag(res)
```

Comment:
@ismagilani Note that those lines 84-93 you removed actually addressed the scenario in which bag_modified flag were false but the bag did not exist. This scenario should not happen, but did happen in HydroShare from time to time, which may have resulted from the error-out of bag creation process which somehow erased the zipped bag file. With your updated bagit process where bag is created in Django, this scenario may not happen, but you never know. It is always good to include those code just as a precaution to fix the potential erroneous scenario by setting bag_modified flag to true if the zipped bag does not exist so that the bag can be recreated. So let's not remove those precaution code.

Reply:
@ismagilani Your consolidated code looks good to me. You don't have to create another PR. You just need to push your updated code to your branch, then github will update this same PR automatically to reflect your updates in your branch.